quickstart/quickstart.cc

```diff
@@ -31,7 +31,7 @@ int main(int argc, char* argv[]) try {
       client.ExecuteQuery(spanner::SqlStatement("SELECT 'Hello World'"));
 
   for (auto const& row : spanner::StreamOf<std::tuple<std::string>>(rows)) {
-    if (!row) throw std::runtime_error(row.status().message());
+    if (!row) throw google::cloud::RuntimeStatusError(row.status());
```

Comment:
> I just don't recall us considering this, unless I've forgotten. Is that the direction you think we should go?

That's the idea, yes.  So I guess this is "consider this" impetus.

> Now, I can certainly be convinced that we do want to make this change and encourage callers to explicitly throw RuntimeStatusError in their own code.

I can be convinced otherwise too.  But, in the context of a `Commit()` mutator as least, anyone taking a `Status` and converting it into throwing (say) a `std::runtime_error` will be defeating the re-run support.  `value()` isn't the only such case.  So our examples probably shouldn't do that (which is what led me here).  And if we chose `RuntimeStatusError` in one context, we might as well just do it everywhere.

As for quickstart, my take is that this doesn't really increase its complexity.  If we really wanted to be simple, we'd get rid of all the throwing/catching and just use `value()` instead.

Reply:
> I can be convinced otherwise too. But, in the context of a `Commit()` mutator as least, anyone taking a `Status` and converting it into throwing (say) a `std::runtime_error` will be defeating the re-run support. `value()` isn't the only such case. So our examples probably shouldn't do that (which is what led me here).

Do we do that anywhere? I don't see any places where we throw from a commit mutator, but maybe I missed it.

Reply:
Observation:

```
if (!row) throw google::cloud::RuntimeStatusError(row.status());
```

is the same as:

```
row.value();
```

I am on record stating that I think the second is better (since we are using exceptions anyway in the examples).  Maybe what we should show is:

```
if (!row) return std::move(row).status();
```

That is, we only throw for application-level errors, and we just "return" to pass library errors back to the library.

Reply:
> Do we do that anywhere? I don't see any places where we throw from a commit mutator, but maybe I missed it.

Yeah, I think you're right.  In the context of `Commit()` mutators, I only see `return`s.  [So now I'm confused about what actually led me here.]

So, unless I hear otherwise, I'll drop this and create a new PR for the few non-`RuntimeStatusError` bits.

Thanks.